File: Client/lib/models/event.dart
class Event {
  final String id;
  final String title;
  final DateTime uploadDate;
  final String location;
  final String description;
  final String banner;
  final String organizerId;
  final List<String>? participants;

  Event({
    required this.id,
    required this.title,
    required this.uploadDate,
    required this.location,
    required this.description,
    required this.banner,
    required this.organizerId,
    this.participants,
  });

  factory Event.fromJson(Map<String, dynamic>? json) {
    return Event(
      id: json?['id'] ?? '',
      title: json?['title'] ?? '',
      uploadDate: DateTime.tryParse(json?['uploadDate'] ?? '') ?? DateTime.now(),
      location: json?['location'] ?? '',
      description: json?['description'] ?? '',
      banner: json?['banner'] ?? '',
      organizerId: json?['organizerId'] ?? '',
      participants: json?['participants'] != null
          ? List<String>.from(json?['participants'] as List<dynamic>)
          : null,
    );
  }
}

File: Server/models/event.go
// models/event.go
package models

import "time"

type Event struct {
	ID           string    `json:"id"`
	Title        string    `json:"title"`
	Date         time.Time `json:"Date"`
	Location     string    `json:"location"`
	Description  string    `json:"description"`
	Banner       string    `json:"banner"`
	OrganizerID  string    `json:"organizerID"`
	Participants []string  `json:"participants"`
}
